src/is_wire/rpc/interceptor.py

```diff
@@ -0,0 +1,6 @@
+class Interceptor(object):
```

Comment:
Why not use an [abstract class](https://docs.python.org/3/library/abc.html)? And why the interfaces of `before_call` and `after_call` differ on `Interceptor` and `LogInterceptor` class?

Reply:
Users can derive from the Interceptor base class if they want (All of our interceptors do). However, it is not necessary to do so. As described by the TypeError raised in the add_interceptor method. The provided interceptor class can implement only the interceptor "concept" and it should be fine. I will improve docstrings though.